extensions/panache/hibernate-orm-panache/deployment/src/main/java/io/quarkus/hibernate/orm/panache/deployment/PanacheHibernateResourceProcessor.java

```diff
@@ -194,4 +202,31 @@ void build(CombinedIndexBuildItem index,
         }
         return null;
     }
+
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    void persistenceUnits(PanacheHibernateOrmRecorder recorder,
+            Optional<JpaEntityPerPersistenceUnitsBuildItem> entityPerPersistenceUnits,
+            List<PanacheEntityClassBuildItem> entityClasses) {
+        Map<String, String> panacheEntityToPersistenceUnit = new HashMap<>();
+        if (entityPerPersistenceUnits.isPresent()) {
+            Map<String, List<String>> collectedEntityToPersistenceUnits = entityPerPersistenceUnits.get()
+                    .getEntityToPersistenceUnits();
+            for (Map.Entry<String, List<String>> entry : collectedEntityToPersistenceUnits.entrySet()) {
+                String entityName = entry.getKey();
+                List<String> selectedPersistenceUnits = entry.getValue();
+                boolean isPanacheEntity = entityClasses.stream()
+                        .anyMatch(entity -> entity.get().name().toString().equals(entityName));
+
+                if (selectedPersistenceUnits.size() > 1 && isPanacheEntity) {
+                    throw new IllegalStateException(String.format(
+                            "PanacheEntity '%s' cannot be defined for usage in 'several persistence unit. The following persistence units were found %s'",
```

Comment:
Also it's not very clear in the message that this is not supported.

Reply:
Added a few words about this.